js/src/elements/form-behavior.ts

```diff
@@ -230,6 +230,7 @@ export class FormBehaviorElement extends HTMLElement {
 			const request = new Request(beforeSubmitEvent.detail.url, beforeSubmitEvent.detail.init);
 			const response = await fetch(request);
 			if (response.ok) {
+				isLoading = true;
```

Comment:
let's remove this line and use the navigate call as the source of truth as to whether a navigation is in progress. We might be able to use the beforeunload event but let's not complicate things too much right now.

```diff
- this.navigate(response.headers.get('location') ?? this.getAttribute('navigation-href'));
+ isNavigating = this.navigate(response.headers.get('location') ?? this.getAttribute('navigation-href'));
```
To make this work, two other changes would be needed:
1. this.navigate return the result of the navigateAfterSubmit invocation.
2. navigateAfterSubmit needs to return true when it navigates:

```diff
export function navigateAfterSubmit(href: string | null, navigationStep: NavigationSteps) {
	switch (navigationStep) {
		case null:
			// do nothing.
-			break;
+     return false;
		case 'follow':
			if (href) {
				location.href = href;
+       return true;
			}
-			break;
+     return false;
		case 'hash-reload':
			if (href) {
				const search = href.includes('?') ? '' : window.location.search;
				if (href !== search + window.location.hash) {
					const state = (history.state || {}) as Record<string, any>;
					window.history.pushState(state, document.title, window.location.pathname + search + href); // prevents scrolling to spot until reload
				}
				location.reload();
+       return true;
			}
-			break;
+     return false;
		case 'replace':
			if (href) {
				location.replace(href);
+       return true;
			}
-			break;
+     return false;
		case 'reload':
			location.reload();
-			break;
+     return true;
		default:
			throw new Error(`Unexpected navigation attribute value.`);
	}
}
```

Reply:
cc @disimmon @homing1 - suggesting this because it looks like navigateAfterSubmit is the only place we can really know whether a navigation is really going to occur at the moment.